src/webgl/interaction.js

```diff
@@ -49,59 +71,81 @@ p5.prototype.orbitControl = function(sensitivityX, sensitivityY) {
-    var deltaTheta = -sensitivityX * (this.mouseX - this.pmouseX) / scaleFactor;
-    var deltaPhi = sensitivityY * (this.mouseY - this.pmouseY) / scaleFactor;
-
-    // camera position
-    var camX = this._renderer.cameraX;
-    var camY = this._renderer.cameraY;
-    var camZ = this._renderer.cameraZ;
-
-    // center coordinates
-    var centerX = this._renderer.cameraCenterX;
-    var centerY = this._renderer.cameraCenterY;
-    var centerZ = this._renderer.cameraCenterZ;
-
-    var diffX = camX - centerX;
-    var diffY = camY - centerY;
-    var diffZ = camZ - centerZ;
-
-    // get spherical coorinates for current camera position about origin
-    var camRadius = Math.sqrt(diffX * diffX + diffY * diffY + diffZ * diffZ);
-    // from https://github.com/mrdoob/three.js/blob/dev/src/math/Spherical.js#L72-L73
-    var camTheta = Math.atan2(diffX, diffZ); // equatorial angle
-    var camPhi = Math.acos(Math.max(-1, Math.min(1, diffY / camRadius))); // polar angle
-
-    // add mouse movements
-    camTheta += deltaTheta;
-    camPhi += deltaPhi;
-
-    // prevent rotation over the zenith / under bottom
-    if (camPhi > Math.PI) {
-      camPhi = Math.PI;
-    } else if (camPhi <= 0) {
-      camPhi = 0.001;
+    if (this.mouseButton === this.LEFT) {
+      var scaleFactor = this.height < this.width ? this.height : this.width;
+      var deltaTheta =
+        -sensitivityX * (this.mouseX - this.pmouseX) / scaleFactor;
+      var deltaPhi = sensitivityY * (this.mouseY - this.pmouseY) / scaleFactor;
+
+      var diffX = cam.eyeX - cam.centerX;
+      var diffY = cam.eyeY - cam.centerY;
+      var diffZ = cam.eyeZ - cam.centerZ;
+
+      // get spherical coorinates for current camera position about origin
+      var camRadius = Math.sqrt(diffX * diffX + diffY * diffY + diffZ * diffZ);
+      // from https://github.com/mrdoob/three.js/blob/dev/src/math/Spherical.js#L72-L73
+      var camTheta = Math.atan2(diffX, diffZ); // equatorial angle
+      var camPhi = Math.acos(Math.max(-1, Math.min(1, diffY / camRadius))); // polar angle
+
+      // add mouse movements
+      camTheta += deltaTheta;
+      camPhi += deltaPhi;
+
+      // prevent rotation over the zenith / under bottom
+      if (camPhi > Math.PI) {
+        camPhi = Math.PI;
+      } else if (camPhi <= 0) {
+        camPhi = 0.001;
+      }
+
+      // from https://github.com/mrdoob/three.js/blob/dev/src/math/Vector3.js#L628-L632
+      // var sinPhiRadius = Math.sin(camPhi) * camRadius;
+      var _x = Math.sin(camPhi) * camRadius * Math.sin(camTheta);
+      var _y = Math.cos(camPhi) * camRadius;
+      var _z = Math.sin(camPhi) * camRadius * Math.cos(camTheta);
+
+      this.camera(
+        _x + cam.centerX,
+        _y + cam.centerY,
+        _z + cam.centerZ,
+        cam.centerX,
+        cam.centerY,
+        cam.centerZ,
+        0,
+        1,
+        0
+      );
     }
+    if (this.mouseButton === this.RIGHT) {
```

Comment:
this isn't if/else—does something strange happen if the user presses both buttons at once?

Reply:
For whatever reason, it just uses the most recently pressed mouseButton behavior with my mouse, but I've switched to `else if` behavior just in case.